src/finalisers/esm.ts

```diff
@@ -110,17 +110,20 @@ function getImportBlock(dependencies: ChunkDependencies, _: string): string[] {
 	return importBlock;
 }
 
-function getExportBlock(exports: ChunkExports, _: string): string[] {
+function getExportBlock(exports: ChunkExports, _: string, varOrConst: string): string[] {
 	const exportBlock: string[] = [];
 	const exportDeclaration: string[] = [];
 	for (const specifier of exports) {
+		let local = specifier.local;
 		if (specifier.exported === 'default') {
-			exportBlock.push(`export default ${specifier.local};`);
+			exportBlock.push(`export default ${local};`);
 		} else {
+			if (specifier.auxLocal) {
+				exportBlock.push(`${varOrConst} ${specifier.auxLocal}${_}=${_}${local};`);
+				local = specifier.auxLocal;
```

Comment:
Is there any logic that makes sure that this new variable does not conflict with an existing one? Worst case, it could conflict with an accessed global variable, in which case the used specifier CANNOT be identical with the property name. In all other cases, it would still need to be deconflicted with all imports and other top-level variables. Unfortunately I do not see a trivial solution for this yet.